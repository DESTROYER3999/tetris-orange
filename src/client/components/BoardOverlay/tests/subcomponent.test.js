import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import BoardOverlaySub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<LoginSub />", () => {
  test("BoardOverlaySub rendering with a player in game", () => {
    const wrapper = shallow(
      <BoardOverlaySub
        hasLost={false}
        isStarted={true}
        inGame={true}
        isHost={false}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("BoardOverlaySub rendering when a player has lost", () => {
    const wrapper = shallow(
      <BoardOverlaySub
        hasLost={true}
        isStarted={true}
        inGame={false}
        isHost={false}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("BoardOverlaySub rendering when in menu and player is host", () => {
    const wrapper = shallow(
      <BoardOverlaySub
        hasLost={false}
        isStarted={false}
        inGame={false}
        isHost={true}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("BoardOverlaySub rendering when in menu and player is not host", () => {
    const wrapper = shallow(
      <BoardOverlaySub
        hasLost={false}
        isStarted={false}
        inGame={false}
        isHost={false}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("BoardOverlaySub rendering when a player has won (hasLost = false and inGame = true)", () => {
    const wrapper = shallow(
      <BoardOverlaySub
        hasLost={false}
        isStarted={false}
        inGame={true}
        isHost={false}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
