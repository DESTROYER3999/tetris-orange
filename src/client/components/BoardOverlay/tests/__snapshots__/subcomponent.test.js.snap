// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> BoardOverlaySub rendering when a player has lost 1`] = `
<styled.div
  id="overlay"
>
  <styled.h1>
    GAME OVER
  </styled.h1>
</styled.div>
`;

exports[`<LoginSub /> BoardOverlaySub rendering when a player has won (hasLost = false and inGame = true) 1`] = `
<styled.div
  id="overlay"
>
  <styled.h1>
    YOU WIN
  </styled.h1>
  <Controls />
  <h2
    id="start-game"
  >
    Wait for host to start
  </h2>
</styled.div>
`;

exports[`<LoginSub /> BoardOverlaySub rendering when in menu and player is host 1`] = `
<styled.div
  id="overlay"
>
  <Controls />
  <h2
    id="start-game"
  >
    Press enter to start
  </h2>
</styled.div>
`;

exports[`<LoginSub /> BoardOverlaySub rendering when in menu and player is not host 1`] = `
<styled.div
  id="overlay"
>
  <Controls />
  <h2
    id="start-game"
  >
    Wait for host to start
  </h2>
</styled.div>
`;

exports[`<LoginSub /> BoardOverlaySub rendering with a player in game 1`] = `
<styled.div
  id="overlay"
/>
`;
